module/zfs/dmu.c

```diff
@@ -1229,7 +1228,9 @@ dmu_object_remap_indirects(objset_t *os, uint64_t object,
 	/*
 	 * Find the next L1 indirect that is not a hole.
 	 */
-	while (dnode_next_offset(dn, 0, &offset, 2, 1, 0) == 0) {
+	rw_enter(&dn->dn_struct_rwlock, RW_WRITER);
```

Comment:
Does this need to be taken as `RW_WRITER`, prior to this change it was `RW_READER`.  Though, I suppose it no longer matters since 59ec30a3290d865b98e3805df6c82b6b4f88aab0 has been merged which removes the `zfs remap` logic.